runtime/src/comm/gasnet/comm-gasnet.c

```diff
@@ -875,7 +875,7 @@ void chpl_comm_exit(int all, int status) {
 }
 
 void  chpl_comm_put(void* addr, c_nodeid_t node, void* raddr,
-                    int32_t elemSize, int32_t typeIndex, int32_t len,
+                    size_t elemSize, int32_t typeIndex, size_t len,
                     int ln, c_string fn) {
   const int size = elemSize*len;
```

Comment:
This should probably be size_t also, though obviously it's not part of this external interface change and C's coercion rules will handle it.